qiling/extensions/pipe.py

```diff
@@ -84,3 +101,82 @@ def flush(self) -> None:
 
     def writable(self) -> bool:
         return True
+
+class SimpleBufferedStream(TextIO):
+    """Simple buffered IO.
+    """
+
+    def __init__(self):
+        self.buff = bytearray()
+        self.cur = 0
+
+    def lseek(self, offset: int, origin: int) -> int:
+        if origin == 0: # SEEK_SET
+            base = 0
+        elif origin == 1: # SEEK_CUR
+            base = self.cur
+        else: # SEEK_END
+            base = len(self.buff) - 1
+
+        if base + offset >= len(self.buff):
+            self.cur = base + offset - 1
+        else:
+            self.cur = base + offset
+
+        return self.cur
+
+    def seek(self, offset: int, origin: int) -> int:
+        return self.lseek(offset, origin)
+    
+    def tell(self) -> int:
+        return self.cur
+
+    def read(self, n: int = -1) -> bytes:
+        if n == -1:
+            ret = self.buff[self.cur:]
+            self.cur = len(self.buff)
+        else:
+            ret = self.buff[self.cur:self.cur + n]
+
+            if self.cur + n >= len(self.buff) - 1:
+                self.cur = len(self.buff)
+            else:
+                self.cur = self.cur + n
+
+        return bytes(ret)
+
+    def readline(self, limit: int = -1) -> bytes:
+        ret = bytearray()
+
+        while not (ret.endswith(b'\n') or len(ret) == limit):
+            ret.extend(self.read(1))
+
+        return bytes(ret)
+
+    def write(self, s: bytes) -> int:
+        if self.cur == 0:
+            self.buff.extend(s)
+        else:
+            self.buff[:self.cur].extend(s)
+        
+        self.cur += len(s)
+
+        return len(s)
+
+    def flush(self) -> None:
+        pass
+
+    def writable(self) -> bool:
+        return True
+
+    def readable(self) -> bool:
+        return True
+
+    def seekable(self) -> bool:
+        return True
+    
+    # This method simply extends internal buffer with self.cur untouched.
+    def extend(self, s: bytes) -> int:
```

Comment:
This is also not part of the `TextIO` interface, and it is unclear where it is actually being used.

Reply:
For example, when you would like to append some bytes in the script, using `write` would change the current position which is easy to cause errors hard to notice.

Reply:
I am not sure that justifies breaking the interface. The idea behind the `Simple` name prefix is to follow the expected interface in the simplest manner. Any addition that breaks this should be implemented elsewhere; those little additions and hacks accumulate quickly to a `DefinitleyNotSimple` class...

Reply:
> I am not sure that justifies breaking the interface. The idea behind the `Simple` name prefix is to follow the expected interface in the simplest manner. Any addition that breaks this should be implemented elsewhere; those little additions and hacks accumulate quickly to a `DefinitleyNotSimple` class...

Okay, make sense.

Reply:
Removed.